core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -538,21 +538,29 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
     val stack = _stack
     val size  = _stackSize // racy
 
-    builder += _lastTrace
+    var last = _lastTrace
```

Comment:
Much more performant to keep the `last` variable in the stack memory instead of checking it in `StackTraceBuilder` which kept it in the heap